docs/05-engineering/git.md

```diff
@@ -0,0 +1,168 @@
+
+- In Sprint 2 work continues on master, v2.0.0 tag gets pushed to stage etc.
+- Meanwhile, critical bugfix needed on prod, so the v1.3.0 is branched to v1.
+  - There is no need for further suffixes in the branch name, since it's the only branch for sprint 1
+  - We don't typically create these branches until they are needed, since it's hopefully rare to need critical hotfixes and we have a tag documenting the state of the release.
+- Fix is pushed to master (i.e. upstream-first workflow), then cherry-picked to v1.
+- Tag v1.3.1 is then created from the v1 branch, pushed to stage (or some dedicated hotfix stage environment) and then to prod.
+- If another hotfix is needed, that commit would also be merged to master, cherry-picked into v1 and tagged v1.3.2 etc.
+
+In a continuous deployment project these tags can be created automatically, or some other mechanism may be used to track releases.
+
+## Git configuration
+
+Here is a recommended Git configuration block that it is recommended to add to your `.
+
+```
+##
+## These are standard CivicActions git settings for developers.
+##
+[core]
+  # This is for MacOS or Linux. For Windows set to `true`.
+  autocrlf = input
+[merge]
+  ff = only
+[pull]
+  ff = only
+```
+
+## GitLab, GitHub, and Bitbucket
+
+GitLab, GitHub, and Bitbucket are services that provides remote access to Git repositories. In addition to hosting your code, these services provide additional features designed to help manage the software development lifecycle. These additional features include managing the sharing of code between different people, bug tracking, wiki space and other tools for "social coding".
+
+- GitHub is a publicly available, free service which requires all code (unless you have a paid account) be made open. Anyone can see code you push to GitHub and offer suggestions for improvement. GitHub currently hosts the source code for tens of thousands of open source projects. CivicActions uses GitHub for all open source projects (except those which are better hosted on an open source community infrastructure), as well as internal, client or pro-bono projects that should be developed in public.
+
+- GitLab is an open source GitHub like software that organizations can use to provide internal management of Git repositories. CivicActions has setup a GitLab server for internal and client projects that should not be developed in public. <https://git.civicactions.net>
+
+### Can Code move between GitLab, GitHub, and Bitbucket?
+
+Yes! All you have to do is point or rebase your local copy to the new remote source. Please note, only your source code will move. Any additional features you use (wiki, bug tracking, etc.) will need to be manually copied.
+
+### When do we use GitLab vs. GitHub?
+
+CivicActions uses GitHub for all open source projects (except those which are better hosted on an open source community infrastructure), as well as internal, client or pro-bono projects that should be developed in public.
```

Comment:
This gets confusing as we are starting to use gitlab.com, but probably too much detail for this document